OMEdit/OMEditGUI/Modeling/LibraryTreeWidget.cpp

```diff
@@ -3967,6 +3968,25 @@ bool LibraryWidget::saveModelicaLibraryTreeItemOneFile(LibraryTreeItem *pLibrary
       pLibraryTreeItem->getModelWidget()->setModelFilePathLabel(fileName);
     }
     mpLibraryTreeModel->updateLibraryTreeItem(pLibraryTreeItem);
+    /* Stage the file for the next commit. */
+    if(MainWindow::instance()->getGitCommands()->isSavedUnderGitRepository(pLibraryTreeItem->getFileName())){
+    QMessageBox *pMessageBox = new QMessageBox(this);
+    pMessageBox->setWindowTitle(QString(Helper::applicationName).append(" - ").append("Stage File"));
+    pMessageBox->setIcon(QMessageBox::Question);
+    pMessageBox->setAttribute(Qt::WA_DeleteOnClose);
+    pMessageBox->setText("Do you want to stage the file for the next commit");
+    pMessageBox->setStandardButtons(QMessageBox::Yes | QMessageBox::No);
+    pMessageBox->setDefaultButton(QMessageBox::Yes);
+    int answer = pMessageBox->exec();
+    switch (answer) {
+      case QMessageBox::Yes:
+        MainWindow::instance()->getGitCommands()->stageCurrentFileForCommit(pLibraryTreeItem->getFileName());
+        break;
+      case QMessageBox::No:
+      default:
+        break;
+    }
+   }
```

Comment:
I guess this code should go into `LibraryWidget::saveFile` since `LibraryWidget::saveModelicaLibraryTreeItemOneFile` is only called for Modelica classes.